test/framework/src/main/java/org/elasticsearch/test/rest/yaml/section/DoSection.java

```diff
@@ -301,10 +301,14 @@ void checkWarningHeaders(final List<String> warningHeaders, final Version master
                      * This warning header will come back in the vast majority of our tests that create an index when running against an
                      * older master. Rather than rewrite our tests to assert this warning header, we assume that it is expected.
                      */
-                } else {
-                    if (expected.remove(message) == false) {
-                        unexpected.add(header);
-                    }
+                } else // noinspection StatementWithEmptyBody
+                    if (message.startsWith("[types removal]")) {
+                    /*
+                     * We skip warnings related to types deprecation so that we can continue to run the many
+                     * mixed-version tests that used typed APIs.
+                     */
+                } else if (expected.remove(message) == false) {
```

Comment:
Shouldn't this be indented a level so the close bracket lines up with the `if` it relates to?

Reply:
I don’t think so, think of it as a multi-line `if`:

```
 if (a
     || b
     || c) {
   // do something
 }
```

Reply:
Ignore me. We don't indent chains of `else if`s and this just threw me because the `else` is separated from the `if` by a comment.